### PR TITLE
Shim from HTTP V1 library to coreHTTP (Serializer only)

### PR DESCRIPTION
**Pull request by yourslab**
Description
-----------

Provides backwards compatibility for the existing API by updating `IotHttpsClient_AddHeader` and `IotHttpsClient_InitializeRequest` to call `HTTPClient_AddHeader` and `HTTPClient_InitializeRequestHeaders` respectively. Another follow-up PR will update `IotHttpsClient_SendSync`, `IotHttpsClient_SendAsync`, and `IotHttpsClient_WriteRequestBody` to call `HTTPClient_Send`.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/c_sdk/standard/https/src/iot_https_client.c [hunk 1]
```diff
@@ -2579,6 +2579,44 @@ static IotHttpsReturnCode_t _initializeResponse( IotHttpsResponseHandle_t * pRes
 
 /*-----------------------------------------------------------*/
 
+static IotHttpsReturnCode_t _shimConvertStatus( HTTPStatus_t coreHttpStatus )
```
Comment (sarenameas) [inline libraries/c_sdk/standard/https/src/iot_https_client.c:2582]: Will you document this?

File: libraries/c_sdk/standard/https/src/iot_https_client.c [hunk 2]
```diff
@@ -619,6 +619,13 @@ static IotHttpsReturnCode_t _initializeResponse( IotHttpsResponseHandle_t * pRes
                                                  IotHttpsResponseInfo_t * pRespInfo,
                                                  _httpsRequest_t * pHttpsRequest );
 
+/**
+ * @brief Convert a status code from coreHTTP to an equivalent status code in the HTTP V1 Library.
+ *
+ * @param[in] coreHttpStatus - The status code from coreHTTP.
```
Comment (sarenameas) [inline libraries/c_sdk/standard/https/src/iot_https_client.c:625]: nitpick, @return

**Review by sarenameas — DISMISSED** [review]

File: libraries/c_sdk/standard/https/src/iot_https_client.c [hunk 3]
```diff
@@ -527,23 +527,22 @@ static IotHttpsReturnCode_t _flushHttpsNetworkData( _httpsConnection_t * pHttpsC
 /**
  * @brief Task pool job routine to send the HTTP request within the pUserContext.
  *
- * @param[in] pTaskPool Pointer to the system task pool.
- * @param[in] pJob Pointer the to the HTTP request sending job.
- * @param[in] pUserContext Pointer to an HTTP request, passed as an opaque context.
+ * @param[in] pTaskPool - Pointer to the system task pool.
+ * @param[in] pJob - Pointer the to the HTTP request sending job.
```
Comment (gkwicker) [inline libraries/c_sdk/standard/https/src/iot_https_client.c:531]: grammar: _Pointer __to the__ HTTP..._

**Review by gkwicker — CHANGES_REQUESTED** [review]
One small grammar fix, see comments

File: libraries/c_sdk/standard/https/src/iot_https_client.c (hunk 3 above)
Comment (yourslab) [reply libraries/c_sdk/standard/https/src/iot_https_client.c:531]: Thanks for catching

**Review by sarenameas — DISMISSED** [review]

**Review by gkwicker — DISMISSED** [review]

**Review by gkwicker — APPROVED** [review]

**Review by sarenameas — APPROVED** [review]

**Pull request merged**